src/main/java/com/puppycrawl/tools/checkstyle/StatelessModule.java

```diff
@@ -0,0 +1,34 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+/**
+ * This interface means that the module does not contain mutable state and
+ * may be safely used from separate threads at the same time.
+ * A module does not contain mutable state, if it does not change fields value at the execution
+ * stage (but it still can set properties at the initialization stage). The best practice is to use
```

Comment:
So this marker has a catch about the `initialization stage` which I didn't see on any of the other javadocs.
Should we include that notice in the others too?

Reply:
Added note to the FileStatefulCheck

Not sure if we should add this note to GlobalStatefulModule, because it does not force check to be immutable

Reply:
> GlobalStatefulModule

It sounds fine that it doesn't include any information on the initialization stage.